### ISSUE TITLE
[$500] [MEDIUM] Expense - Preview loads infinitely from admin view when member deletes request with receipt

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#33953]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.21-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. [Member] Go to workspace chat.
2. [Member] Create a manual request with receipt.
3. [Member] Delete the request.
4. [Admin] Go to the workspace chat with member. (Do not open workspace chat with employee until the request is deleted).

## Expected Result:
The expense is removed from the workspace chat.

## Actual Result:
The expense is not removed and the receipt thumbnail loads infinitely.
On Android and iOS, no infinite loading, just a blank preview.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/7a2ab147-c428-4dbe-a77f-17080ebfe86e

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01468ff4e2ac20d0b7</li>
        <li>Upwork Job ID: 1742893423100641280</li>
        <li>Last Price Increase: 2024-01-18</li>
    </ul>
</details>

**Comment by zPilax** [comment]
Hi Team, just a suggestion for this one:

I think your delete code on that button is just deleting the file/img, you might want to compare the codes from add to your delete code, maybe you'll find something there.

**Comment by samilabud** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Preview loads infinitely from admin view when member deletes request with receipt.

### What is the root cause of that problem?
We do not take into consideration when an image does not exist and gives an error because of this: https://github.com/Expensify/App/blob/8aec07c55df3f0fefc6bed76c991183cf6b97f7a/src/components/ImageWithSizeCalculation.tsx#L80

### What changes do you think we should make in order to solve the problem?
We should set a value at the reference (isLoadedRef) when we get an error inside this function:
https://github.com/Expensify/App/blob/8aec07c55df3f0fefc6bed76c991183cf6b97f7a/src/components/ImageWithSizeCalculation.tsx#L46
E.G:
```
const onError = () => {
        isLoadedRef.current = true;
        Log.hmmm('Unable to fetch image to calculate size', {url});
    };
```
Before:

https://github.com/Expensify/App/assets/5216036/f76e4471-7db6-4be5-8ed1-308aa6cd2686


After:

https://github.com/Expensify/App/assets/5216036/b11ce72b-ea0b-4f1f-8f50-d98e039cd0d7


Knowing this we can add a functionality to hide extra space or maybe show a custom warning.

**Comment by aim-salam** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Expense - Preview loads infinitely from admin view when member deletes request with receipt

### What is the root cause of that problem?

`<ReportPreview` didnt listen to `isHidden`.

https://github.com/Expensify/App/blob/095161a690e0cea4cf5352b3d03212530a364fea/src/pages/home/report/ReportActionItem.js#L138

https://github.com/Expensify/App/blob/095161a690e0cea4cf5352b3d03212530a364fea/src/pages/home/report/ReportActionItem.js#L338-L348


### What changes do you think we should make in order to solve the problem?

Add `isHidden` prop into `ReportPreview`

```js
                <ReportPreview
                    isHidden={isHidden} //<=======
                    iouReportID={ReportActionsUtils.getIOUReportIDFromReportActionPreview(props.action)}
                    chatReportID={props.report.reportID}
                    policyID={props.report.policyID}
                    containerStyles={props.displayAsGroup ? [] : [styles.mt2]}
                    action={props.action}
                    isHovered={hovered}
                    
                    contextMenuAnchor={popoverAnchorRef}
                    checkIfContextMenuActive={toggleContextMenuFromActiveReportAction}
                    isWhisper={isWhisper}
                />
```

And wrap entire component with `isHidden` if-else statement in `ReportPreview.js`



### What alternative solutions did you explore? (Optional)

**Comment by aim-salam** [comment]
.

**Comment by aim-salam** [comment]
backend issue. Issue related with API `DeleteMoneyRequest`

**Comment by kevinksullivan** [comment]
Hey @Ollyws , what's your ETA on reviewing? Thanks!

**Comment by samilabud** [comment]
> @samilabud Thanks for the proposal but the problem doesn't seem to be the spinner itself, but rather the fact that the deleted request is still show.

Thanks for checking out, that is definitely the problem, because of that the alternative proposal could be taken to show a warning saying the request was deleted or maybe hide the whole post.

**Comment by samilabud** [comment]
> @samilabud Thanks for the update, but why is the report preview not being removed when deleted?

Currently when we remove the the money request from user A, for user B we are displaying the amount from the action message:
https://github.com/Expensify/App/blob/497ed9273d99afd1cdb061acb78014e15ffdefe8/src/components/ReportActionItem/ReportPreview.js#L205
Because of that we has the perception that we are seeing the same message/report that was deleted, but the truth is the report doesn't exist.

**Comment by samilabud** [comment]
Hi @Ollyws,

After my answer yesterday I checked further, I realized the root cause is due of how the way push notification works. 

I mean while we are using the expensify app we receive updates which are handled by [OnyxUpdates](https://github.com/Expensify/App/blob/main/src/libs/actions/OnyxUpdates.ts), these updates works perfect for the same level of thread but in this case we are deleting the money request in the sub-thread (after clicking the IOU and then using context menu and selecting Delete request):
<img width="607" alt="image" src="https://github.com/Expensify/App/assets/5216036/18931b12-273d-4d64-b8ba-444303bf03aa">

What happens then? Well all the other expensify app that are opened receive the update for the sub-thread but not for the main one, we storage this in Onyx and this is the real reason why we keep seeing the request, if you clear the Onyx data manually this fixed.

That said I have to say that so far I don't know how to fix the push notification so that it also updates the parent of the deleted request, if you have any ideas I can try.
